lib/vanagon/component/dsl.rb

```diff
@@ -184,7 +184,7 @@ def install_service(service_file, default_file = nil, service_name = @component.
           # Return here because there is no file to install, just a string read in
           return
         when "windows"
-          @component.service = OpenStruct.new(:name => service_name, :service_file => service_file)
+          @component.service = OpenStruct.new(:name => service_name, :id => "#{service_name.gsub(/([^A-Za-z0-9])/, '').upcase}", :service_file => target_service_file)
```

Comment:
I think the merge has gone in, so I'll do the target_XXX fix as MAINT once the windows/vanagon call is over.